Gems/Atom/RPI/Code/Include/Atom/RPI.Edit/Material/MaterialTypeSourceData.h

```diff
@@ -97,6 +109,58 @@ namespace AZ
                 MaterialPropertyValue m_softMin;
                 MaterialPropertyValue m_softMax;
                 MaterialPropertyValue m_step;
+
+            private:
+
+                // We are gradually moving toward having a more proper API for MaterialTypeSourceData code, but we still some public members
+                // like above. However, it's important for m_name to be private because it is used as the key for lookups, collision validation, etc.
+                AZStd::string m_name; //!< The name of the property within the property group. The full property ID will be groupName.propertyName.
+            };
+            
+            using PropertyList = AZStd::vector<AZStd::unique_ptr<PropertyDefinition>>;
+
+            struct PropertyGroup
+            {
+                friend class MaterialTypeSourceData;
+                
+                AZ_CLASS_ALLOCATOR(PropertyGroup, SystemAllocator, 0);
+                AZ_TYPE_INFO(AZ::RPI::MaterialTypeSourceData::PropertyGroup, "{BA3AA0E4-C74D-4FD0-ADB2-00B060F06314}");
+
+            public:
+
+                PropertyGroup() = default;
+                AZ_DISABLE_COPY(PropertyGroup)
+
+                const AZStd::string& GetName() const { return m_name; }
+                const AZStd::string& GetDisplayName() const { return m_displayName; }
+                const AZStd::string& GetDescription() const { return m_description; }
+                const PropertyList& GetProperties() const { return m_properties; }
+                const AZStd::vector<AZStd::unique_ptr<PropertyGroup>>& GetPropertyGroups() const { return m_propertyGroups; }
+                const AZStd::vector<Ptr<MaterialFunctorSourceDataHolder>>& GetFunctors() const { return m_materialFunctorSourceData; }
+                
+                void SetDisplayName(AZStd::string_view displayName) { m_displayName = displayName; }
+                void SetDescription(AZStd::string_view description) { m_description = description; }
```

Comment:
I'm not sure if there are public-facing coding standards for O3DE. But yes, before we released to open source, I believe one-line accessor functions were allowed like this.

Reply:
It turns out that O3DE does have public-facing coding standards: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md

All I could find on the matter seems to indicate that it's required to avoid single line statements:

**Single statement per line**
**Required**: Each line should contain no more than a single statement.

**Reason**: It is easy to misread lines with multiple statements compounded together. It also helps with debugging in some tools e.g. breakpoint placement in Visual Studio is per line, not per statement.

````
// Unacceptable: Single line function definition
bool BadMethod() { return false; }

// Acceptable: Method/function definition with the body on its own line (following bracing convention)
bool GoodMethod()
{
    return false;
}
````

I'll ask in the next sig-core meeting specifically about class' getters and setters since it'd help if the documentation would have a section specifically about those.

Reply:
I'll get it in another PR.